File: Task/Write-float-arrays-to-a-text-file/C++/write-float-arrays-to-a-text-file-2.cpp
#include <algorithm>
#include <cmath>    // ::sqrt()
#include <fstream>
#include <iomanip>  // setprecision()
#include <iostream>
#include <string>
#include <vector>

int main()
{
  try {
    // prepare test data
    double x[] = {1, 2, 3, 1e11};
    const size_t xsize = sizeof(x) / sizeof(*x);
    std::vector<double> y(xsize);
    std::transform(&x[0], &x[xsize], y.begin(), ::sqrt);

    // write file using default precisions
    writedat("sqrt.dat", &x[0], &x[xsize], y.begin(), y.end());

    // print the result file
    std::ifstream f("sqrt.dat");
    for (std::string line; std::getline(f, line); )
      std::cout << line << std::endl;
  }
  catch(std::exception& e) {
    std::cerr << "writedat: exception: '" << e.what() << "'\n";
    return 1;
  }
  return 0;
}

File: Task/Write-float-arrays-to-a-text-file/Python/write-float-arrays-to-a-text-file-2.py
>>> import math
>>> x = [1, 2, 3, 1e11]
>>> y = map(math.sqrt, x)
>>> y
[1.0, 1.4142135623730951, 1.7320508075688772, 316227.76601683791]
>>> writedat("sqrt.dat", x, y)
>>> # check
...
>>> for line in open('sqrt.dat'):
...   print line,
...
1       1
2       1.4142
3       1.7321
1e+011  3.1623e+005
